### PR TITLE
you can press the conveyor switch in hand to link any not deployed belts to it

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
:cl:
add: you can press the conveyor switch in hand to link any not deployed belts to it
/:cl:

### CODE REVIEW

File: code/modules/recycling/conveyor2.dm [hunk 1]
```diff
@@ -370,6 +370,13 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 	. = ..()
 	id = "[rand()]" //this couldn't possibly go wrong
 
+/obj/item/conveyor_switch_construct/attack_self(mob/user)
+	var/counter = 0
+	for(var/obj/item/conveyor_construct/C in view())
```
Comment (Qustinnus) [inline code/modules/recycling/conveyor2.dm]: 😩

**Comment by Dawson1917** [comment]
>for(var/obj/item/conveyor_construct/C **in view())**

nearly had me worried there

File: code/modules/recycling/conveyor2.dm [hunk 2]
```diff
@@ -370,6 +370,13 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
...
+		C.id = id
+		counter++
+	to_chat(user, "<span class='notice'>You have linked [counter] conveyor belts to this switch.</span>")
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/recycling/conveyor2.dm]: 1 conveyor belts

File: code/modules/recycling/conveyor2.dm [hunk 3]
```diff
@@ -370,6 +370,13 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 	. = ..()
 	id = "[rand()]" //this couldn't possibly go wrong
 
+/obj/item/conveyor_switch_construct/attack_self(mob/user)
+	var/counter = 0
+	for(var/obj/item/conveyor_construct/C in view())
+		C.id = id
+		counter++
+	to_chat(user, "<span class='notice'>You have linked [counter ? "conveyor belts" : "conveyor belt"] to this switch.</span>")
```
Comment (nemvar) [inline code/modules/recycling/conveyor2.dm]: That's not how this works.

File: code/modules/recycling/conveyor2.dm [hunk 4]
```diff
@@ -370,6 +370,11 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 	. = ..()
 	id = "[rand()]" //this couldn't possibly go wrong
 
+/obj/item/conveyor_switch_construct/attack_self(mob/user)
+	for(var/obj/item/conveyor_construct/C in view())
+		C.id = id
+	to_chat(user, "<span class='notice'>You have linked all nearby converyor belt assemblies to this switch.</span>")
```
Comment (nemvar) [inline code/modules/recycling/conveyor2.dm]: conveyor not converyor

**Comment by ExcessiveUseOfCobblestone** [comment]
does `view` care about things like darkness/blindness/etc.? 

If so prob just want to do a set range (smaller is better) since that seems really unintuitive.

**Comment by Qustinnus** [comment]
@ExcessiveUseOfCobblestone its a byond proc so no

**Pull request merged**